### PR TITLE
Modules for ALB, NLB & corresponding listeners

### PR DESCRIPTION
**Pull request by ringods**
Fix for: https://github.com/skyscrapers/core/issues/12

* Module `alb_listener` between `alb` and `alb_rule_target`. Makes it easier to manage target groups, listeners and listener rules. 
* Modules `nlb` and `nlb_listener`.

### CODE REVIEW

**Review by MattiasGees — APPROVED** [review]
Look ok, for me this is a major version bump

**Review by venturel — APPROVED** [review]
It looks good also to me. I agree with Mattias on the major version bump

**Review by SamClinckspoor — COMMENTED** [review]
is there any value in the nlb and alb parts? I can understand for the listener but the main aws_lb modules don't do much

File: nlb/main.tf [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Create a new load balancer
+resource "aws_lb" "nlb" {
+  load_balancer_type         = "network"
+  name                       = "${var.project}-${var.environment}-${var.name}-nlb"
```
Comment (SamClinckspoor) [inline nlb/main.tf]: please use `name_prefix`

File: alb/main.tf [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # Create a new load balancer
-resource "aws_alb" "alb" {
+resource "aws_lb" "alb" {
+  load_balancer_type         = "application"
   name                       = "${var.project}-${var.environment}-${var.name}-alb"
```
Comment (SamClinckspoor) [inline alb/main.tf]: please use `name_prefix`

File: nlb/sg.tf [hunk 3]
```diff
@@ -0,0 +1,11 @@
+resource "aws_security_group" "sg_nlb" {
+  name        = "sg_alb_${var.project}_${var.environment}_${var.name}"
```
Comment (SamClinckspoor) [inline nlb/sg.tf]: please use `name_prefix`

**Review by SamClinckspoor — CHANGES_REQUESTED** [review]

**Comment by ringods** [comment]
@SamClinckspoor the reason is cost. Contrary to an ELB, we will no longer set up a application or network load balancer per application. The LB will be a shared resource.

For an application load balancer, we add HTTP and/or HTTPS listeners, then add `aws_lb_listener_rule` with different target groups to route to different hosts.

For a network load balancer, we add additional port specific listeners.

**Comment by SamClinckspoor** [comment]
I mean, what value does the module bring. As far as I can tell the `alb` and `nlb` don't provide a lot of abstraction to make it easier to use compared to using the resource instead of the module

**Review by SamClinckspoor — APPROVED** [review]

**Comment by ringods** [comment]
It's debatable indeed, but having the modules (lb + sg) makes the setup consistent module wise.

**Pull request merged**